src/PrayerTimes.js

```diff
@@ -26,11 +34,36 @@ export default class PrayerTimes {
         sunriseTime = new TimeComponents(solarTime.sunrise).utcDate(date.getFullYear(), date.getMonth(), date.getDate());
         var sunsetTime = new TimeComponents(solarTime.sunset).utcDate(date.getFullYear(), date.getMonth(), date.getDate());
 
+        const polarCircleResolver = calculationParameters.polarCircleResolution;
+        if (
+          (!isValidDate(sunriseTime) || !isValidDate(sunsetTime))
+          && polarCircleResolver !== PolarCircleResolution.unresolved
+        ) {
+            const resolved = polarCircleResolvedValues(polarCircleResolver, date, coordinates);
+            this.coordinates = resolved.coordinates;
+            this.date.setTime(resolved.date.getTime());
+            solarTime = resolved.solarTime;
+            const dateComponents = [date.getFullYear(), date.getMonth(), date.getDate()];
+
+            dhuhrTime = new TimeComponents(solarTime.transit).utcDate(...dateComponents);
+            sunriseTime = new TimeComponents(solarTime.sunrise).utcDate(...dateComponents);
+            sunsetTime = new TimeComponents(solarTime.sunset).utcDate(...dateComponents);
+        }
+
         asrTime = new TimeComponents(solarTime.afternoon(shadowLength(calculationParameters.madhab))).utcDate(date.getFullYear(), date.getMonth(), date.getDate());
 
         var tomorrow = dateByAddingDays(date, 1);
         var tomorrowSolarTime = new SolarTime(tomorrow, coordinates);
+        if (
+          (isNaN(tomorrowSolarTime.sunrise))
+          && polarCircleResolver !== PolarCircleResolution.unresolved
+        ) {
+            const resolved = polarCircleResolvedValues(polarCircleResolver, tomorrow, coordinates);
```

Comment:
This concerns me slightly, as tomorrowSunrise needs to actually be one day later than the sunrise value. It's used for some high latitude adjustments like 1/7 of the night calculations. With the aqrab yaum calculation it might go back to the same day that the original resolver arrived at.

Perhaps we should move the first invalid check down here and then check if either sunrise, sunset or tomorrowSunrise are invalid. Then we can have the resolver find values that will allow the date and the following date to have valid sunrise and sunset times.

Reply:
If my understanding is correct, we must ensure that the tomorrow's solar time will precisely correspond to the solar time of the day after the one we resolved for the today's solar time.
Thus, what we need in the resolution process, is to ensure that for the resolved today's solar time, the solar time of the day after is also valid. I made the changes correspondingly to that understanding.